File: java/main/src/pl/vot/tomekby/World/Plants/Belladonna.java
package pl.vot.tomekby.World.Plants;

import pl.vot.tomekby.World.Organism;
import pl.vot.tomekby.World.World;

/**
 * Created by Tomasz on 2015-05-07.
 */
final public class Belladonna extends Plant {
    public Belladonna() { super(); init(); }
    public Belladonna(int x, int y) { super(x, y); init(); }
    public Belladonna(int x, int y, World world) { super(x, y, world); init(); }

    public String get_name() { return "Wilcze jagody"; }
    public void collision(Organism with) {
        // Zabicie zwierzęcia które zjadło roślinę
        kill(with);
    }
    protected void spawn_new(int x, int y) {
        super.spawn_new(new Belladonna(x, y));
    }
    protected void init() {
        strength = 99;
        SYMBOL = 'j';
    }
}

File: py/World/Belladonna.py
from World.Plant import Plant


class Belladonna(Plant):
    def get_name(self):
        return "Wilcze jagody"

    # Specyficzna akcja przy kolizji - zabicie organizmu który zjadł
    def collision(self, second):
        self._kill(second)

    # Inicjalizacja
    def _init(self):
        self._SYMBOL = 'j'
        self._strength = 99
